pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -249,6 +249,9 @@ func getPodRevision(pod *v1.Pod) string {
 // newStatefulSetPod returns a new Pod conforming to the set's Spec with an identity generated from ordinal.
 func newStatefulSetPod(set *apps.StatefulSet, ordinal int) *v1.Pod {
 	pod, _ := controller.GetPodFromTemplate(&set.Spec.Template, set, metav1.NewControllerRef(set, controllerKind))
+	if pod == nil {
```

Comment:
This is not correct. If the function returns nil we should panic. That code path should never occur.